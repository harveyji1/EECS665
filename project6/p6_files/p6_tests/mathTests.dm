main : () void{
    myB : int = 2;
    myA : int = 1;
    myC : int;
    myD : int;
    myC = myA + myB;
    myD = myC * 3;
    myB = myD / 3;
    myA = myB - 2;
}