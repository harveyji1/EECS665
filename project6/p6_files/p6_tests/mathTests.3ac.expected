[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN main LOCALS]
myB (local var of 8 bytes)
myA (local var of 8 bytes)
myC (local var of 8 bytes)
myD (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
tmp3 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            [tmp0] := [myA] ADD64 [myB]
            [myC] := [tmp0]
            [tmp1] := [myC] MULT64 3
            [myD] := [tmp1]
            [tmp2] := [myD] DIV64 3
            [myB] := [tmp2]
            [tmp3] := [myB] SUB64 2
            [myA] := [tmp3]
lbl_0:      leave main

